src/views/Generator/Generator.hooks.ts

```diff
@@ -41,7 +37,10 @@ export function useUpdateValueInGeneratorFile(fileName: string) {
   return useMutation({
     mutationFn: async ({ key, value }: { key: string; value: unknown }) => {
```

Comment:
Nothing to do with this PR really, but just an idea: instead of passing a key and value to update (which doesn't seem to be type safe?), why not use a `map` function:

```ts
useUpdateInFile(fileName: (generator: Generator) => {
  return {
    ...generator,
    myValue: "hello"
  }
})
```